cas-client-core/src/main/java/org/jasig/cas/client/validation/AbstractTicketValidationFilter.java

```diff
@@ -186,6 +196,26 @@ protected void onFailedValidation(final HttpServletRequest request, final HttpSe
         // nothing to do here.
     }
 
+    private void regenerateSession(HttpServletRequest request) {
+        HttpSession oldSession = request.getSession();
+        Enumeration<String> attrNames = oldSession.getAttributeNames();
+        Properties props = new Properties();
+
+        if (attrNames != null) {
```

Comment:
Yes, but it will be always at least one attribute: the assertion with the ticket validation. But perhaps it will be better to access to old and new session and then copy the attributes. It will be clearer.